### PR TITLE
feat(linter): implement no-global-assign

### LINKED ISSUE #598: feat(linter): eslint/no-global-assign
Implement ESLint's [`no-global-assign`](https://eslint.org/docs/latest/rules/no-global-assign) rule.

Start by running
```
just new-rule no-global-assign
```

### PR DESCRIPTION
**Pull request by SoonIter**
closes #598

### CODE REVIEW

**Review by Boshen — APPROVED** [review]
We can revisit the commented out test cases if they come up in real codebases.

File: crates/oxc_linter/src/rules/eslint/no_global_assign.rs [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+                .unwrap_or(&vec![])
+                .iter()
+                .map(serde_json::Value::as_str)
+                .filter(std::option::Option::is_some)
+                .map(|x| Atom::from(x.unwrap().to_string()))
+                .collect::<Vec<Atom>>(),
+        }
+    }
+
+    fn run<'a>(&self, _node: &oxc_semantic::AstNode<'a>, ctx: &LintContext<'a>) {
```
Comment (Boshen) [inline crates/oxc_linter/src/rules/eslint/no_global_assign.rs:55]: This is run on every ast node, you can use the `run_on_symbol` function.

Comment (Boshen) [inline crates/oxc_linter/src/rules/eslint/no_global_assign.rs:55]: This is run on every ast node, you can use the run_on_symbol function.

**Review by Boshen — CHANGES_REQUESTED** [review]

File: crates/oxc_linter/src/rules/eslint/no_global_assign.rs (hunk 1 above)
Comment (SoonIter) [reply crates/oxc_linter/src/rules/eslint/no_global_assign.rs:55]: OK, I feel that there are two problems here.
1. It seems that global variables that is not resolved can not be executed by "run_on_symbol". 🤔 I may need some help in this place
2. global variables should be divided into different environments such as browser or node as eslint test cases shown. [eslint-test-case](https://github.com/eslint/eslint/blob/853d32baa8934c08b59a738470b72522e1505f6f/tests/lib/rules/no-global-assign.js#L27C8-L27C57)

Comment (Boshen) [reply crates/oxc_linter/src/rules/eslint/no_global_assign.rs:55]: For 1, let me add a `run_once` function.
For 2, we can revisit this when we have a "Environments" option.

Comment (Boshen) [reply crates/oxc_linter/src/rules/eslint/no_global_assign.rs:55]: @SoonIter Checkout https://github.com/web-infra-dev/oxc/pull/647/files

**Review by Boshen — APPROVED** [review]

**Pull request merged**